apps/src/templates/progress/progressHelpers.js

```diff
@@ -118,6 +121,54 @@ export function stageIsAllAssessment(levels) {
   return levels.every(level => level.kind === LevelKind.assessment);
 }
 
+/**
+ * Computes progress status percentages for a set of levels.
+ * @param {{id:studentLevelProgressType}} studentProgress An object keyed by
+ * level id containing objects representing the student's progress in that level
+ * @param {levelType[]} levels An array of levels
+ * @returns {object} An object with a percentage of levels in each of the
+ * following buckets: attempted, imperfect, completed, incomplete.
+ *
+ * Note: this function will replace `summarizeProgressInStage` below once we
+ * refactor the legacy StudentProgressSummaryCell component
+ */
+export function statusPercentsForLevels(studentProgress, levels) {
+  // Filter any bonus levels as they do not count toward progress.
+  const filteredLevels = levels.filter(level => !level.bonus);
+  const statuses = filteredLevels.map(level => {
+    const levelProgress = studentProgress[level.id];
+    return (levelProgress && levelProgress.status) || LevelStatus.not_tried;
+  });
+
+  const completedStatuses = [
+    LevelStatus.perfect,
+    LevelStatus.submitted,
+    LevelStatus.free_play_complete,
+    LevelStatus.completed_assessment,
+    LevelStatus.readonly
```

Comment:
through the "manage locking" ux for assessments and surveys. it means the student can view the assessment but not edit it (as opposed to "locked", which doesn't allow student to view at all).

Reply:
Ah, I see.  Is it possible for a read-only assessment to not be completed?  Or do we always want to show a read-only assessment as being completed?

Reply:
that's a good question, but one for amanda. the scope of this work was intended to achieve feature parity with our existing tables, so unless things looked broken i tried not to change the design at all. but read-only is "complete" in the sense that the student can't do anything more with it, so maybe the "complete-colored" part of the box is intended to show how much of the lesson is "done" rather than "complete"?